sdks/java/extensions/jackson/src/test/java/org/apache/beam/sdk/extensions/jackson/JacksonTransformsTest.java

```diff
@@ -0,0 +1,242 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.jackson;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.collect.Iterables;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.SerializableCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * Test Jackson transforms {@link ParseJsons} and {@link AsJsons}.
+ */
+public class JacksonTransformsTest {
+  private static final List<String> VALID_JSONS =
+      Arrays.asList(
+          "{\"myString\":\"abc\",\"myInt\":3}",
+          "{\"myString\":\"def\",\"myInt\":4}"
+      );
+
+  private static final List<String> INVALID_JSONS =
+      Arrays.asList(
+          "{myString:\"abc\",\"myInt\":3,\"other\":1}",
+          "{",
+          ""
+      );
+
+  private static final List<String> EMPTY_JSONS =
+      Arrays.asList(
+          "{}",
+          "{}"
+      );
+
+
+  private static final List<String> EXTRA_PROPERTIES_JSONS =
+      Arrays.asList(
+          "{\"myString\":\"abc\",\"myInt\":3,\"other\":1}",
+          "{\"myString\":\"def\",\"myInt\":4}"
+      );
+
+  private static final List<MyPojo> POJOS =
+      Arrays.asList(
+          new MyPojo("abc", 3),
+          new MyPojo("def", 4)
+      );
+
+  private static final List<MyEmptyBean> EMPTY_BEANS =
+      Arrays.asList(
+          new MyEmptyBean("abc", 3),
+          new MyEmptyBean("def", 4)
+      );
+
+  @Rule
+  public final transient TestPipeline pipeline = TestPipeline.create();
+
+  @Test
+  public void parseValidJsons() {
+    PCollection<MyPojo> output =
+        pipeline
+            .apply(Create.of(VALID_JSONS))
+            .apply(ParseJsons.of(MyPojo.class)).setCoder(SerializableCoder.of(MyPojo.class));
+
+    PAssert.that(output).containsInAnyOrder(POJOS);
+
+    pipeline.run();
+  }
+
+  @Test(expected = Pipeline.PipelineExecutionException.class)
+  public void failParsingInvalidJsons() {
+    PCollection<MyPojo> output =
+        pipeline
+            .apply(Create.of(Iterables.concat(VALID_JSONS, INVALID_JSONS)))
+            .apply(ParseJsons.of(MyPojo.class)).setCoder(SerializableCoder.of(MyPojo.class));
```

Comment:
On one of my other tests, someone recommended I put all of the strings and objects for the test right in the test itself. I did and it made the tests more readable.